lib/connextclient/ConnextClient.ts

```diff
@@ -455,6 +457,20 @@ class ConnextClient extends SwapClient {
             preimage: transferStatusResponse.preImage?.slice(2),
           };
         case 'EXPIRED':
+          const expiredTransferUnlocked$ = defer(() => from(this.settleInvoice(rHash, '', currency))).pipe(
```

Comment:
I'm a bit confused as to what this is doing (some comments might be good). Why are we "settling" a payment where we are the sender? I know that "invoice" is not exactly a connext term/concept, maybe we should just use the connext call/terminology directly here rather than `settleInvoice` which is a generic SwapClient method.

Reply:
When the connext transfer (HTLC) expires the funds are not automatically returned to the channel balance. Expired transfer apps can be unlocked by both the receiver and sender by calling `/hashlock-resolve` with the `paymentId`.

In this PR the `settleInvoice` function has been modified to always include the `paymentId` in the body of the request in case the transfer has been expired.

It's worth noting that the node also periodically cleans up the expired transfer apps, but from UX perspective it's better if this happens from xud side. This ensures that the funds locked in the transfer app are available to the user sooner.

I've added a short comment above the `settleInvoice` call.

> maybe we should just use the connext call/terminology directly here rather than settleInvoice which is a generic SwapClient method

Are you suggesting to create a wrapper function here that internally calls `settleInvoice`?

Reply:
OK, thanks for the explanation this makes sense. I think it would be great to put a brief summary of this in a line or two of comments right above this call.

I think calling `settleInvoice` here with a blank preimage is still a bit odd, I would just go ahead and call `/hashlock-resolve` directly like so:

```
await this.sendRequest('/hashlock-resolve', 'POST', {
  assetId,
  paymentId: sha256(['address', 'bytes32'], [assetId, `0x${rHash}`]),
});
```

And the settleInvoice call can be left as it currently is unless you think the payment id is useful to include even when we have the preimage?

Reply:
Updated.

> And the settleInvoice call can be left as it currently is unless you think the payment id is useful to include even when we have the preimage?

Yep, removed payment id from settleInvoice.